pkg/minikube/constants/constants.go

```diff
@@ -81,6 +81,24 @@ func GetMinishiftHomeDir() string {
 	if ok {
 		return homeEnv
 	} else {
-		return filepath.Join(util.HomeDir(), ".minishift")
+		return filepath.Join(GetHomeDir(), ".minishift")
 	}
 }
+
+// GetHomeDir returns the home directory for the current user
+func GetHomeDir() string {
+	if runtime.GOOS == "windows" {
+		if homeDrive, homePath := os.Getenv("HOMEDRIVE"), os.Getenv("HOMEPATH"); len(homeDrive) > 0 && len(homePath) > 0 {
+			homeDir := homeDrive + homePath
+			if _, err := os.Stat(homeDir); err == nil {
```

Comment:
I believe the line which constructs the path on Windows is weak (and error prone), as there is no enforcement of format. The stat does check if exists... but there are no tests for this.

Reply:
@gbraad This method was moved here as it is from `pkg/util/homedir.go`

Did you mean the the line `homeDir := homeDrive + homePath` should be formed using `filepath.Join` instead?

Reply:
if possible, yes

Reply:
@praveenkumar Tested in windows working fine!

Reply:
I also run the PR on Windows 10. `minishift start` worked fine.